Assets/Script/Projectile.cs

```diff
@@ -1,27 +1,44 @@
-using System.Collections;
 using UnityEngine;
 
 [RequireComponent(typeof(Rigidbody))]
 public class Projectile : MonoBehaviour
 {
     private Rigidbody rb;
-    private float velocity;
+    [SerializeField] private float velocity;
+    [SerializeField] private float damage;
+    [SerializeField] private GameObject explosion;
+
     private void Start()
     {
         Setup();
         StartMoving();
+        Destroy(gameObject, 5f);
     }
     private void Setup()
     {
         rb = gameObject.GetComponent<Rigidbody>();
     }
     public void StartMoving()
     {
-        rb.velocity = gameObject.transform.forward * 50f;
+        //rb.AddForce(gameObject.transform.forward * velocity, ForceMode.Impulse);
+        rb.velocity = gameObject.transform.forward * velocity;
     }
 
-    private void OnCollisionEnter(Collision collision)
+    public void Explode()
     {
+        GameObject vfx = Instantiate(explosion, gameObject.transform.position, Quaternion.identity);
+        vfx.transform.parent = null;
+        Destroy(vfx.gameObject, 2);
         Destroy(gameObject);
     }
+
+    private void OnCollisionEnter(Collision collision)
+    {
+        Debug.Log($"Damage Taken: {damage}, {collision.gameObject.name}");
+        if (collision.gameObject.GetComponentInParent<Damageable>() != null)
```

Comment:
You should use TryGetComponent instead of GetComponent for these cases where the object might not have the component.

https://docs.unity3d.com/ScriptReference/Component.TryGetComponent.html